packages/neon/neon/lib/src/widgets/drawer.dart

```diff
@@ -0,0 +1,210 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:meta/meta.dart';
+import 'package:neon/l10n/localizations.dart';
+import 'package:neon/neon.dart';
+import 'package:neon/src/router.dart';
+import 'package:neon/src/widgets/drawer_destination.dart';
+import 'package:provider/provider.dart';
+
+@internal
+class NeonDrawer extends StatelessWidget {
+  const NeonDrawer({
+    super.key,
+  });
+
+  @override
+  Widget build(final BuildContext context) {
+    final accountsBloc = Provider.of<AccountsBloc>(context, listen: false);
+    final appsBloc = accountsBloc.activeAppsBloc;
+
+    return StreamBuilder(
+      stream: appsBloc.appImplementations,
+      builder: (final context, final snapshot) {
+        if (snapshot.data?.data == null) {
+          return Container();
+        }
+
+        return _NeonDrawer(
+          apps: snapshot.data!.data!,
+        );
+      },
+    );
+  }
+}
+
+class _NeonDrawer extends StatefulWidget {
+  const _NeonDrawer({
+    required this.apps,
+  });
+
+  final Iterable<AppImplementation> apps;
+
+  @override
+  State<_NeonDrawer> createState() => __NeonDrawerState();
+}
+
+class __NeonDrawerState extends State<_NeonDrawer> with SingleTickerProviderStateMixin {
+  late TabController _tabController;
+  late AccountsBloc _accountsBloc;
+  late AppsBloc _appsBloc;
+  late List<AppImplementation> _apps;
+
+  int _activeApp = 0;
+
+  @override
+  void initState() {
+    super.initState();
+
+    _accountsBloc = Provider.of<AccountsBloc>(context, listen: false);
+    _appsBloc = _accountsBloc.activeAppsBloc;
+
+    _apps = widget.apps.toList();
+    _activeApp = _apps.indexWhere((final app) => app.id == _appsBloc.activeAppID.valueOrNull);
+
+    _tabController = TabController(
+      vsync: this,
+      length: widget.apps.length,
+    );
+  }
+
+  @override
+  void dispose() {
+    _tabController.dispose();
+    super.dispose();
+  }
+
+  void onAppChange(final int index) {
+    setState(() {
+      _activeApp = index;
+    });
```

Comment:
I already expected this to break when I saw this change. When you go to the settings item it correctly opens the settings page, but afterwards the settings item is selected. I guess you can just move it to the end of the method to fix this.

Reply:
I only tested it with the normal drawer and totally forgot the persistent one.